### ISSUE TITLE
Bug: Different empty `alt` values for `image`, `imageReference`

### ISSUE THREAD

**Issue by yaycmyk** [wooorm/remark#103]
using `mdast.parse()`:

```js
mdast.parse("![](/xyz.png)")
```
Yields an AST node for image with `alt` set to `null`.

The same setup but using an imageReference instead:

```js
mdast.parse("![][1]\n[1]: /xyz.png")
```

Yields an imageReference with `alt` set to `""` (empty string)

**Comment by wooorm** [comment]
I agree, that isn’t very pretty. I’d go for the following [here](https://github.com/wooorm/mdast/blob/2f2761609f24dc0bb0ea41421a8b8848bc9cc715/lib/parse.js#L1798):

```diff
 } else if (type === 'image') {
-    node.alt = decode(self.descape(text), eat);
+    node.alt = decode(self.descape(text), eat) || null;
 }
```

The reason for `null` is because [it’s in the `imageReference` definition](https://github.com/wooorm/mdast/blob/master/doc/mdastnode.7.md#imagereference).

Not that I suggest accept both `null` and `undefined` next to `""`, everywhere. It makes for more solid code. Although you’re building with mdast ingrained right now, maybe in the future you’ll allow plug-ins to manipulate the tree too, in which case they might (probably will) sometimes forget to add a property.

**Comment by wooorm** [comment]
This will be included in the upcoming release of mdast, 3.0.0, which I’ll expect to :ship: before the new year.

**Comment by yaycmyk** [comment]
:tada:
